corporate/lib/decorator.py

```diff
@@ -104,6 +105,18 @@ def _wrapped_view_func(
                 query = urlencode({"next_page": page_type})
                 url = append_url_query_string(url, query)
 
+            # Return error for AJAX requests with url.
+            if request.headers.get("x-requested-with") == "XMLHttpRequest":  # nocoverage
+                return HttpResponse(
+                    orjson.dumps(
+                        {
+                            "error_message": "Remote billing authentication expired",
+                            "login_url": url,
+                        }
+                    ),
+                    status=401,
+                )
```

Comment:
This is not a good way to check if it's an API request; we should instead be determining from the endpoints themselves whether it's an API request or a normal page. And this code path is already sloppy on that -- we return a 404.

This should work ok for now; we can fix further in a follow-up project that makes this decorator know what sort of endpoint it decorates and thus what error format to use in a cleaner way.

@mateuszmandera FYI in case you see a good way to do this; simplest I see on a quick look is having this decorator take an `api` vs. `page` parameter.

Reply:
@timabbott Yeah I think that sounds like a good approach. Or should we be actually using a namespace in the endpoint path to separate api vs user-facing page?

Reply:
A namespace in the path would not be a bad idea for readability; maybe post a recommendation in Zulip?